contractcourt/chain_watcher.go

```diff
@@ -790,12 +806,27 @@ func (c *chainWatcher) handleKnownRemoteState(
 		return false, nil
 	}
 
+	// Create an AnchorResolution for the breached state.
+	anchorRes, err := lnwallet.NewAnchorResolution(
+		c.cfg.chanState, commitSpend.SpendingTx,
+	)
+	if err != nil {
+		return false, fmt.Errorf("unable to create anchor "+
+			"resolution: %v", err)
+	}
+
+	// We'll set the ConfCommitKey here as the remote htlc set. This is
+	// only used to ensure a nil-pointer-dereference doesn't occur and is
```

Comment:
If I got this doc right, this can be dangerous if we are setting a value just to get around the nil pointer issue. Shouldn't we just check against nil?

Reply:
That is an alternative, but `InsertConfirmedCommitSet` will need a `ConfCommitKey` when encoding the commit set. This would require making a new `ConfCommitKey` since 3/4 are used we can "make" another one. IMO seems like more effort than is necessary as the callpath later on just uses `RemoteHtlcSet` in `prepContractResolutions` which calls `constructChainActions` which the breach case ignores. No codepath currently uses `nil` in the `ConfCommitKey` so it would only be dangerous if a code-path used `nil`.

Reply:
This is correct though AFAICT: we're handling a breach, which means the remote party force closed, which means we only have one main commitment to play.

Reply:
@Crypt-iQ thanks for the detailed explanation! My main concern is that we are setting a value where it shouldn't, as later on other callers may access it while it doesn't fulfill what it claims to be, thus increasing the cognitive load. Looks like it's the very ending of the code-path so I'd figure it should be fine, plus based on @Roasbeef 's comment I think the `ConfCommitKey` is indeed the `RemoteHtlcSet`?